src/components/Calculator.js

```diff
@@ -1,58 +1,49 @@
-import React from 'react';
+import React, { useState } from 'react';
 import '../App.css';
 import './Calculator.css';
 import calculate from '../logic/calculate';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
+const Calculator = () => {
+  const [state, setState] = useState({
+    total: 0,
+    next: null,
+    operation: null,
+  });
 
-    this.state = {
-      total: 0,
-      next: null,
-      operation: null,
-    };
-
-    this.handleEvent = this.handleEvent.bind(this);
-  }
-
-  handleEvent = (buttonName) => {
-    this.setState((state) => calculate(state, buttonName));
+  const handleEvent = (e) => {
+    setState({ ...state, ...calculate(state, e.target.name) });
   };
 
-  render() {
-    const { total, operation, next } = this.state;
-    return (
-      <div>
-        <p className="result">
-          <span>{total}</span>
-          <span>{operation}</span>
-          <span>{next}</span>
-        </p>
-        <div className="calc">
-          <button type="button" className="grayBtn" name="AC" onClick={(e) => this.handleEvent(e.target.name)}>AC</button>
-          <button type="button" className="grayBtn" name="+/-" onClick={(e) => this.handleEvent(e.target.name)}>+/-</button>
-          <button type="button" className="grayBtn" name="%" onClick={(e) => this.handleEvent(e.target.name)}>%</button>
-          <button type="button" className="orangeBtn" name="÷" onClick={(e) => this.handleEvent(e.target.name)}>÷</button>
-          <button type="button" className="grayBtn" name="7" onClick={(e) => this.handleEvent(e.target.name)}>7</button>
-          <button type="button" className="grayBtn" name="8" onClick={(e) => this.handleEvent(e.target.name)}>8</button>
-          <button type="button" className="grayBtn" name="9" onClick={(e) => this.handleEvent(e.target.name)}>9</button>
-          <button type="button" className="orangeBtn" name="x" onClick={(e) => this.handleEvent(e.target.name)}>x</button>
```

Comment:
- [ ] You refactored the class component into a functional component using React hooks, very well :+1: :clap: . But when an operator button is clicked multiple times consecutively, the browser throws an error. Kindly add a method to handle this error as your app needs to be without browser or console error. :pray: 

![Screenshot from 2021-09-15 20-38-17](https://user-images.githubusercontent.com/64894927/133455012-7de45235-10d1-4040-afd9-e30be382697a.png)